aibtc-v1/crews/contract_code_generator.py

```diff
@@ -0,0 +1,194 @@
+    verbose=True,
+    memory=True,
+    backstory=(
+        "You are a Clarity code compilation expert responsible for taking the generated Clarity code and the code review report, and combining them into a final, comprehensive output that can be displayed in the Streamlit app."
+    ),
+    allow_delegation=False,
+    llm=llm
+)
+
+# Define the task for generating Clarity code
+generate_clarity_code_task = Task(
+    description=(
+        "Generate a Clarity code snippet for a smart contract that defines and manages token transfers, including functions for minting, transferring, and checking balances. The code should ensure security, prevent re-entrancy, and handle exceptions properly."
+        "Store your code in crew's shared memory 'contract_code' "
+
+    ),
+    expected_output="Provide a Clarity code snippet for a smart contract that defines and manages token transfers.",
+    agent=clarity_code_generator,
+)
+
+# Define the task for reviewing Clarity code
+review_clarity_code_task = Task(
+    description=(
+        "Review the generated Clarity code, create a Clarinet project with it, and check its syntax. "
+        "Provide a report on the code quality and any issues found during the syntax check. You can find the code in crew's shared memory 'contract code', store your response in crew's shared memory 'review' "
+    ),
+    expected_output="A detailed report on code quality, syntax check results, and any issues found.",
+    agent=clarity_code_reviewer,
+)
+
+# Define the task for compiling the Clarity code and review report
+compile_clarity_code_task = Task(
+    description=(
+        "Combine the generated Clarity code and the code review report into a comprehensive output that can be displayed in the Streamlit app. You can access the contract code from crew's shared memory 'contract_code' and the code review from 'review'."
+    ),
+    expected_output="The final output that includes the Clarity code and the code review report.",
+    agent=clarity_code_compiler,
+)
+crew = Crew(
+    agents=[clarity_code_generator,
+            clarity_code_reviewer, clarity_code_compiler],
+    tasks=[generate_clarity_code_task,
+           review_clarity_code_task, compile_clarity_code_task],
+    process=Process.sequential,
+    verbose=True
+)
+# # Forming the crew with both agents and their tasks
+# crew = Crew(
+#     agents=[clarity_code_generator, clarity_code_reviewer],
+#     tasks=[generate_clarity_code_task, review_clarity_code_task],
+#     process=Process.sequential,
+#     verbose=True
+# )
+
+# Function to run the crew
+
+
+def generate_and_review_contract(user_input):
+    result = crew.kickoff(inputs={"user_input": user_input})
+    print(result, "result(*(***))")
+
+    return result
+
+# Streamlit app definition
+
+
+def main():
+    st.title("Clarity Smart Contract Generator and Reviewer")
+
+    user_input = st.text_area("Enter your smart contract requirements:",
```

Comment:
I see here the crew kicks off with user input, but I don't see that used later in the tasks? Is the user input considered, or is the task hardcoded to generating a token (as mentioned in earlier comment)?